test/integration/controller/tfjob/tfjob_controller_test.go

```diff
@@ -0,0 +1,723 @@
+						corev1.ResourceCPU: "on-demand",
+					},
+					Count: ptr.To(createdWorkload.Spec.PodSets[0].Count),
+				},
+				kueue.PodSetAssignment{
+					Name: "PS",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "spot",
+					},
+					Count: ptr.To(createdWorkload.Spec.PodSets[1].Count),
+				},
+				kueue.PodSetAssignment{
+					Name: "Worker",
+					Flavors: map[corev1.ResourceName]kueue.ResourceFlavorReference{
+						corev1.ResourceCPU: "spot",
+					},
+					Count: ptr.To(createdWorkload.Spec.PodSets[2].Count),
+				},
+			).
+			Obj()
+		gomega.Expect(util.SetAdmission(ctx, k8sClient, createdWorkload, admission)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+				return false
+			}
+			return !*createdJob.Spec.RunPolicy.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+		gomega.Expect(len(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypeChief].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypeChief].Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(onDemandFlavor.Name))
+		gomega.Expect(len(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypePS].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypePS].Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(spotFlavor.Name))
+		gomega.Expect(len(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypeWorker].Template.Spec.NodeSelector)).Should(gomega.Equal(1))
+		gomega.Expect(createdJob.Spec.TFReplicaSpecs[kftraining.TFJobReplicaTypeWorker].Template.Spec.NodeSelector[labelKey]).Should(gomega.Equal(spotFlavor.Name))
+		gomega.Consistently(func() bool {
+			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+				return false
+			}
+			return len(createdWorkload.Status.Conditions) == 1
+		}, util.ConsistentDuration, util.Interval).Should(gomega.BeTrue())
+
+		ginkgo.By("checking the workload is finished when job is completed")
+		createdJob.Status.Conditions = append(createdJob.Status.Conditions,
+			kftraining.JobCondition{
+				Type:               kftraining.JobSucceeded,
+				Status:             corev1.ConditionTrue,
+				LastTransitionTime: metav1.Now(),
+			})
+		gomega.Expect(k8sClient.Status().Update(ctx, createdJob)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+			if err != nil || len(createdWorkload.Status.Conditions) == 1 {
+				return false
+			}
+
+			return apimeta.IsStatusConditionTrue(createdWorkload.Status.Conditions, kueue.WorkloadFinished)
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+	})
+})
+
+var _ = ginkgo.Describe("Job controller for workloads when only jobs with queue are managed", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
```

Comment:
I would remove this test. It can be tested in unit. And it's already tested for other jobs.

Reply:
Do other jobs mean kubeflowjobs?

Reply:
Other jobs in general. But, I guess we can keep the test for PytorchJob

Reply:
It makes sense.